homeassistant/loader.py

```diff
@@ -884,18 +884,42 @@ def _resolve_integrations_from_root(
 
 
 @callback
-def async_get_loaded_integration(hass: HomeAssistant, domain: str) -> Integration:
+def async_loaded_integration_has_no_requirements(
+    hass: HomeAssistant, domain: str
+) -> bool:
+    """Check if an loaded integration has no requirements.
+
+    If the integration is not loaded, this will return False
+    as we don't know if it has requirements or not.
+    """
+    if integration := _async_get_loaded_integration(hass, domain):
+        return not integration.requirements
+    return False
+
```

Comment:
This needs a direct test

Reply:
Maybe it's better to return them if it has them instead.  That way requirements processor can prime its cache and check against already loaded and probably skip a whole lot more

Reply:
Maybe better to not do this at all and just use it to prime the cache when we start